### PR TITLE
[DOCS] Network direction processor

### PR DESCRIPTION
**Pull request by danhermann**
See the processor added in https://github.com/elastic/elasticsearch/pull/66644

### CODE REVIEW

File: docs/reference/ingest/processors/network-direction.asciidoc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+| `ignore_missing`   | no       | `true`        | If `true` and any required fields are missing,
+the processor quietly exits without modifying the document.
+
+
+include::common-options.asciidoc[]
+|======
+
+[float]
+[[supported-named-network-ranges]]
+===== Supported named network ranges:
```
Comment (jrodewig) [inline docs/reference/ingest/processors/network-direction.asciidoc:35]: I don't think the colon is needed here.
```suggestion
===== Supported named network ranges
```

File: docs/reference/ingest/processors/network-direction.asciidoc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+- `private` - Matches private address ranges defined in RFC 1918 (IPv4) and
+  RFC 4193 (IPv6).
+- `public` - Matches addresses that are not loopback, unspecified, IPv4
+  broadcast, link local unicast, link local multicast, interface local
+  multicast, or private.
+- `unspecified` - Matches unspecified addresses (either the IPv4 address
+  "0.0.0.0" or the IPv6 address "::").
+
+
+The following examples illustrates the use of the network direction processor:
```
Comment (jrodewig) [inline docs/reference/ingest/processors/network-direction.asciidoc:57]: I'd create a separate heading for clarity.
```suggestion

[discrete]
[[network-direction-processor-ex]]
===== Examples

The following examples illustrates the use of the network direction processor:
```

File: docs/reference/ingest/processors/network-direction.asciidoc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        },
+        "destination": {
+          "ip": "192.168.1.1"
+        }
+      }
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
```
Comment (jrodewig) [inline docs/reference/ingest/processors/network-direction.asciidoc:86]: I think we can convert this to a working console example.
```suggestion
[source,console]
----
POST _ingest/pipeline/_simulate
{
  "pipeline": {
    "processors": [
      {
        "network_direction": {
          "internal_networks": ["private"]
        }
      }
    ]
  },
  "docs": [
    {
      "_source": {
        "source": {
          "ip": "128.232.110.120"
        },
        "destination": {
          "ip": "192.168.1.1"
        }
      }
    }
  ]
}
----
```

File: docs/reference/ingest/processors/network-direction.asciidoc [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    "source": {
+      "ip": "128.232.110.120"
+    },
+    "destination": {
+      "ip": "192.168.1.1"
+    },
+    "network": {
+      "direction": "inbound"
+    }
+  }
```
Comment (jrodewig) [inline docs/reference/ingest/processors/network-direction.asciidoc:104]: We can probably convert this to a working `console-result` example as well.

My suggestion does replace the two ellipses with a single one, but I don't think that's a big deal. It also flips `_source.destination` and `_source.source` so they're returned in the same order as the response.

```suggestion
[source,console-result]
----
{
  "docs": [
    {
      "doc": {
        ...
        "_source": {
          "destination": {
            "ip": "192.168.1.1"
          },
          "source": {
            "ip": "128.232.110.120"
          },
          "network": {
            "direction": "inbound"
          }
        }
      }
    }
  ]
}
----
// TESTRESPONSE[s/\.\.\./"_index":"_index","_id":"_id","_ingest":{"timestamp":$body.docs.0.doc._ingest.timestamp},/]
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left a few comments but nothing blocking. Thanks @danhermann.

**Comment by danhermann** [comment]
Thanks, @jrodewig! I incorporated all your suggestions into the doc.

**Pull request merged**